WorkflowSwiftUI/Sources/ObservableState.swift

```diff
@@ -0,0 +1,181 @@
+import Foundation
+import IdentifiedCollections
+
+/// A type that emits notifications to observers when underlying data changes.
+///
+/// Conforming to this protocol signals to other APIs that the value type supports observation.
+/// However, applying the ``ObservableState`` protocol by itself to a type doesn’t add observation
+/// functionality to the type. Instead, always use the ``ObservableState()`` macro when adding
+/// observation support to a type.
+#if !os(visionOS)
+public protocol ObservableState: Perceptible {
+    var _$id: ObservableStateID { get }
+    mutating func _$willModify()
+}
+#else
+public protocol ObservableState: Observable {
+    var _$id: ObservableStateID { get }
+    mutating func _$willModify()
+}
+#endif
+
+/// A unique identifier for a observed value.
+public struct ObservableStateID: Equatable, Hashable, Sendable {
+    @usableFromInline
+    var location: UUID {
+        get { storage.id.location }
+        set {
```

Comment:
Should this bail if the new value is the same as the old value?

Reply:
This is a derived file from TCA. Probably the set isn't expensive enough to matter, would be my assumption.